packages/microcosm/src/microcosm.js

```diff
@@ -109,34 +109,18 @@ class Microcosm extends Emitter implements Domain {
 
     this.options = merge(DEFAULTS, this.constructor.defaults, preOptions || {})
 
+    this.active = true
     this.parent = this.options.parent
-
     this.initial = this.parent ? this.parent.initial : this.getInitialState()
-    this.state = this.parent ? this.parent.state : this.initial
-
+    this.snapshots = Object.create(this.parent ? this.parent.snapshots : null)
     this.history = this.parent ? this.parent.history : new History(this.options)
-
     this.actions = Object.create(this.parent ? this.parent.actions : null)
-    this.snapshots = Object.create(this.parent ? this.parent.snapshots : null)
-    this.domains = new DomainEngine(this)
-    this.effects = new EffectEngine(this)
-    this.changes = new CompareTree(this.state)
-    this.active = true
-
-    // History moves through a set lifecycle. As that lifecycle occurs,
-    // save snapshots of new state:
-
-    // When an action is first created
-    this.history.on('append', this._createSnapshot, this)
+    this.changes = new CompareTree(this.initial)
```

Comment:
We still always need a compare tree per fork. Forks mitigate a lot of the value of the compare tree because we can't take advantage of structural sharing between data subscriptions.

Something to think about...